sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -120,6 +120,54 @@ object ResolveDefaultColumns {
     }
   }
 
+  /**
+   * Computes the string representation of an expression suitable for use when storing the
+   * constant-folded result of DEFAULT column values.
+   */
+  private def expressionToString(expression: Expression): String = {
+    expression match {
+      case Cast(child, dataType, _, _) =>
+        s"CAST(${expressionToString(child)} AS ${dataType.catalogString})"
+      case Literal(value, dataType: DataType) =>
+        valueToString(value, dataType)
+      case _ =>
+        expression.sql
+    }
+  }
+
+  /**
+   * Computes the string representation of a value suitable for use when storing the
+   * constant-folded result of DEFAULT column values.
+   */
+  private def valueToString(value: Any, dataType: DataType): String = {
+    (value, dataType) match {
+      case (data: GenericArrayData, arrayType: ArrayType) =>
```

Comment:
Shall we fix the `sql` method of `Literal` instead?

Reply:
Sure, this sounds good. Done. Thanks for the suggestion!